### PR TITLE
make extension dll depends on merged nuget.exe

### PR DESCRIPTION
**Pull request by zhili1208**

### CODE REVIEW

File: src/NuGet.Clients/NuGet.MSSigning.Extensions/NuGet.MSSigning.Extensions.csproj [hunk 1]
```diff
@@ -11,10 +11,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\..\NuGet.Clients\NuGet.CommandLine\NuGet.CommandLine.csproj" />
-  </ItemGroup>
-
-  <ItemGroup>
+    <Reference Include="NuGet">
```
Comment (rohit21agrawal) [inline src/NuGet.Clients/NuGet.MSSigning.Extensions/NuGet.MSSigning.Extensions.csproj:14]: why does it need to depend on the actual .exe? we  don't recommend people do that

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.MSSigning.Extensions/NuGet.MSSigning.Extensions.csproj:14]: Also, do you really need the Reference includes for system, system.core on SDK-based csproj? (I know it's from an old check-in)

File: build/common.project.props [hunk 2]
```diff
@@ -250,6 +250,9 @@
     <SolutionProjects Include="$(RepositoryRootDirectory)src\*\*\*.csproj" />
     <SolutionProjectsWithoutVSIX Include="@(SolutionProjects)"
                                 Exclude="$(VSIXProject)" />
+    <ExtensionProjects Include="$(RepositoryRootDirectory)src\*\*\*.Extensions.csproj" />
```
Comment (rohit21agrawal) [inline build/common.project.props:253]: reason we don't want it to build?

File: build/build.proj [hunk 3]
```diff
@@ -308,6 +308,16 @@
                          VisualStudioVersion=15.0;" />
   </Target>
 
+  <Target Name="BuildVS15CommandExtensions" AfterTargets= "BuildVS15NoVSIX" Condition=" '$(IsXPlat)' != 'true'">
```
Comment (rohit21agrawal) [inline build/build.proj:311]: why build them separately?

**Comment by zhili1208** [comment]
close it since choose another approach

**Pull request closed without merging**